src/core/public/application/types.ts

```diff
@@ -90,6 +90,27 @@ export enum AppNavLinkStatus {
   hidden = 3,
 }
 
+/**
+ * Visibilities of the application
+ *
+ * @public
+ */
+export enum AppVisibility {
+  /**
+   * The application visibility will be `both` if the application's {@link AppVisibility} is set to `default` or not set
+   * which means the application is visible within / out of workspace.
+   */
+  default = 0,
+  /**
+   * The application is only visible when user is inside a workspace.
+   */
+  workspaceOnly = 1,
+  /**
+   * The application is only visible when user is not in any workspace.
+   */
+  homeOnly = 2,
```

Comment:
Do we use `home` to generalize when user is not in any workspace?

Reply:
Can be rephrased but as we are using workspaceless, we can ignore this concern for now.